src/misc/ssh/ssh_decode_sequence_multi.c

```diff
@@ -0,0 +1,136 @@
+   va_start(args, inlen);
+   for (;;) {
+      type = (ssh_data_type)va_arg(args, int);
+      data = va_arg(args, void*);
+      LTC_UNUSED_PARAM(data);
+
+      if (type == LTC_SSHDATA_EOL) {
+         break;
+      }
+
+      switch (type) {
+         case LTC_SSHDATA_BYTE:
+            size = ((unsigned long)in[0]);
+            in++;
+            *(unsigned long *)data = size;
+            break;
+         case LTC_SSHDATA_BOOLEAN:
+            size = ((unsigned long)in[0]);
+            in++;
+            /*
+               The value 0 represents FALSE, and the value 1 represents TRUE.  All non-zero values MUST be
+               interpreted as TRUE; however, applications MUST NOT store values other than 0 and 1.
+            */
+            *(unsigned long *)data = (size)?1:0;
+            break;
+         case LTC_SSHDATA_UINT32:
+            size = ((unsigned long)in[0] << 24) |
+                   ((unsigned long)in[1] << 16) |
+                   ((unsigned long)in[2] << 8) |
+                   ((unsigned long)in[3]);
+            in += 4;
+            *(unsigned long *)data = size;
+            break;
+         case LTC_SSHDATA_UINT64:
+            size = ((unsigned long)in[0] << 56) |
+                   ((unsigned long)in[1] << 48) |
+                   ((unsigned long)in[2] << 40) |
+                   ((unsigned long)in[3] << 32) |
+                   ((unsigned long)in[4] << 24) |
+                   ((unsigned long)in[5] << 16) |
+                   ((unsigned long)in[6] << 8) |
+                   ((unsigned long)in[7]);
+            in += 8;
+            *(unsigned long *)data = size;
+            break;
+         case LTC_SSHDATA_STRING:
+         case LTC_SSHDATA_NAMELIST:
+            bufsize = va_arg(args, unsigned long);
+            size = ((unsigned long)in[0] << 24) |
+                   ((unsigned long)in[1] << 16) |
+                   ((unsigned long)in[2] << 8) |
+                   ((unsigned long)in[3]);
+            in += 4;
+            if (size > 0) {
+               if (size >= bufsize) {
+                  va_end(args);
+                  return CRYPT_BUFFER_OVERFLOW;
+               }
+               strncpy((char *)data, (const char *)in, size);
+               ((char *)data)[size] = 0; /* strncpy doesn't null terminate */
+            } else {
+               *(char *)data = 0;
+            }
+            in += size;
+            break;
+         case LTC_SSHDATA_MPINT:
+            size = ((unsigned long)in[0] << 24) |
+                   ((unsigned long)in[1] << 16) |
+                   ((unsigned long)in[2] << 8) |
+                   ((unsigned long)in[3]);
+            in += 4;
+            if (size == 0) {
+               if ((err = mp_set(data, 0)) != CRYPT_OK) {
+                  va_end(args);
+                  return CRYPT_ERROR;
+               }
+            } else {
+               if ((err = mp_read_unsigned_bin(data, (unsigned char *)in, size)) != CRYPT_OK) {
```

Comment:
~~I just realized that the `src` argument in `unsigned_read()` in `tomcrypt_math.h` could be changed to `const unsigned char *src` as both ltm and tfm nowadays have `const*` arguments where applicable~~ oops ... ltm isn't released yet with those changes... \*cough\* https://github.com/libtom/libtommath/issues/122 \*cough\*